src/Tizen.NUI/src/public/WindowSystem/TzShell.cs

```diff
@@ -0,0 +1,122 @@
+                    {
+                        throw new NotSupportedException("Not Supported");
+                    }
+                    else if (error == (int)Interop.TzShell.ErrorCode.NoService)
+                    {
+                        throw new InvalidOperationException("No Service");
+                    }
+                    else
+                    {
+                        throw new InvalidOperationException("Unknown Error");
+                    }
+                }
+
+                /// <summary>
+                /// Creates a new Tizen Shell handle.
+                /// </summary>
+                /// <returns>A new Tizen Shell instance.</returns>
+                /// <exception cref="Tizen.Applications.Exceptions.OutOfMemoryException">This exception can be due to out Of memory.</exception>
+                /// <exception cref="ArgumentException">This exception can be due to an invalid state.</exception>
+                public TzShell()
+                {
+                    _tzsh = Interop.TzShell.tzshCreate((int)Interop.TzShell.ToolKitType.Efl);
+                    if (_tzsh == (IntPtr)0)
+                    {
+                        int err = Tizen.Internals.Errors.ErrorFacts.GetLastResult();
+                        ErrorCodeThrow(err);
+                    }
+                }
+
+                /// <summary>
+                /// Destructor.
+                /// </summary>
+                /// <exception cref="ArgumentException">This exception can be due to an invalid state.</exception>
+                ~TzShell()
+                {
+                    Dispose(false);
+                }
+
+                /// <summary>
+                /// Dispose.
+                /// </summary>
+                /// <exception cref="ArgumentException">This exception can be due to an invalid state.</exception>
+                public void Dispose()
+                {
+                    Dispose(true);
+                    GC.SuppressFinalize(this);
+                }
+
+                /// <summary>
+                /// Dispose.
+                /// </summary>
+                /// <exception cref="ArgumentException">This exception can be due to an invalid state.</exception>
+                protected virtual void Dispose(bool disposing)
+                {
+                    int _res;
+                    if (!disposed)
+                    {
+                        if (_tzsh != (IntPtr)0)
+                        {
+                            _res = Interop.TzShell.tzshDestroy(_tzsh);
+                            ErrorCodeThrow(_res);
+                            _tzsh = (IntPtr)0;
+                        }
+                        disposed = true;
+                    }
+                }
+
+                /// <summary>
+                /// Get a tizen ws shell handler.
+                /// </summary>
+                /// <returns>tizen ws shell handler</returns>
+                public IntPtr GetNativeHandle()
```

Comment:
Do not open a method to return the native pointer handle. Please use SafeHandle if necessary.

Reply:
change this API to internal